File: Api.Server/src/Services/Center/Menu/Dto/QueryMenuPagedInput.cs
using Fast.Center.Enum;

namespace Fast.Center.Service.Menu.Dto;

/// <summary>
/// <see cref="QueryMenuPagedInput"/> 获取菜单列表输入
/// </summary>
public class QueryMenuPagedInput : PagedInput
{
    /// <summary>
    /// 版本
    /// </summary>
    public EditionEnum? Edition { get; set; }

    /// <summary>
    /// 应用Id
    /// </summary>
    public long? AppId { get; set; }

    /// <summary>
    /// 模块Id
    /// </summary>
    public long? ModuleId { get; set; }

    /// <summary>
    /// 菜单类型
    /// </summary>
    public MenuTypeEnum? MenuType { get; set; }

    /// <summary>
    /// 是否桌面端
    /// </summary>
    public bool? HasDesktop { get; set; }

    /// <summary>
    /// 是否Web端
    /// </summary>
    public bool? HasWeb { get; set; }

    /// <summary>
    /// 是否移动端
    /// </summary>
    public bool? HasMobile { get; set; }

    /// <summary>
    /// 是否显示
    /// </summary>
    public YesOrNotEnum? Visible { get; set; }

    /// <summary>
    /// 状态
    /// </summary>
    public CommonStatusEnum? Status { get; set; }
}

File: App.Client/src/api/services/menu/models/QueryMenuPagedInput.ts
import { EditionEnum } from "@/api/enums/EditionEnum";
import { MenuTypeEnum } from "@/api/enums/MenuTypeEnum";
import { CommonStatusEnum } from "@/api/enums/CommonStatusEnum";

/**
 * Fast.Center.Service.Menu.Dto.QueryMenuPagedInput 获取菜单列表输入
 */
export interface QueryMenuPagedInput extends PagedInput  {
  /**
   * 
   */
  edition?: EditionEnum;
  /**
   * 应用Id
   */
  appId?: number;
  /**
   * 模块Id
   */
  moduleId?: number;
  /**
   * 
   */
  menuType?: MenuTypeEnum;
  /**
   * 是否桌面端
   */
  hasDesktop?: boolean;
  /**
   * 是否Web端
   */
  hasWeb?: boolean;
  /**
   * 是否移动端
   */
  hasMobile?: boolean;
  /**
   * 是否显示
   */
  visible?: boolean;
  /**
   * 
   */
  status?: CommonStatusEnum;
}

